README.md

```diff
@@ -1 +1,204 @@
-# OpenSLO
\ No newline at end of file
+# OpenSLO
+
+## Table of Contents
+
+- [OpenSLO](#openslo)
+  - [Introduction](#introduction)
+  - [Specification](#specification)
+    - [Goals](#goals)
+    - [Object Types](#object-types)
+      - [General Schema](#general-schema)
+        - [Notes](#notes)
+      - [SLO](#slo)
+        - [Notes](#notes-1)
+        - [Objectives](#objectives)
+      - [Service](#service)
+
+## Introduction
+
+The intent of this document is to outline the OpenSLO specification.
+
+The goal of this project is to provide an open specification for defining and
+interfacing with SLOs to allow for a common, giving a set vendor-agnostic
+solution to defining and tracking SLOs. Platform specific implimentation details
+are purposefully excluded from the scope of this specification.
+
+## Specification
+
+### Goals
+
+- Compliance with the Kubernetes YAML format.
+- Vendor agnostic
+- Be flexible enough to be extended elsewhere
+
+### Object Types
+
+> 💡 **Note:** Specific attributes are described in detail in the **Notes** and
+> under each integration section.
+
+#### General Schema
+
+```yaml
+apiVersion: openslo/v1alpha
+kind: SLO | Service
+metadata:
+  name: string
+  displayName: string   # optional
+spec:
+```
+
+##### Notes
+
+- **kind** *string* - required, either [SLO](#slo) or [Service](#service)
+- **name:** *string* - required field, convention for naming object from
+  [DNS RFC1123](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names)
+  `name` should:
+  - contain at most 63 characters
+  - contain only lowercase alphanumeric characters or `-`
+  - start with an alphanumeric character
+  - end with an alphanumeric character
+
+---
+
+#### SLO
+
+A service level objective (SLO) is a target value or range of values for
+a service level that is measured by a service level indicator (SLI).
+
+```yaml
+apiVersion: n9/v1alpha
+kind: SLO
+metadata:
+  name: string
+  displayName: string #optional
+  project: string
+spec:
+  description: string #optional
+  service: [service name] #name of the service to associate this SLO with
+  timeWindows:
+    # exactly one item, one of possible rolling time window or calendar aligned
+    # rolling time window
+    - unit: Day | Hour | Minute
```

Comment:
I'm going to argue that the denomination should just be seconds here (and elsewhere). There's always weird corner-cases that don't fit into pre-determined boundaries.

Reply:
I agree.  Another future improvement might be to later add support for abbreviated units, like `1h` or `24s`